app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/yeartodatelliability/YearToDateLiabilityController.scala

```diff
@@ -542,24 +552,28 @@ class YearToDateLiabilityController @Inject() (
                 c => hasEstimatedDetailsForm.fill(c.hasEstimatedDetails)
               )
             )(
-              page = hasEstimatedDetailsPage(_, _)
+              page = hasEstimatedDetailsPage(_, _, isATrust(fillingOutReturn))
             )(
-              requiredPreviousAnswer               = _.fold(_.taxableGainOrLoss, c => Some(c.taxableGainOrLoss)),
+              requiredPreviousAnswer = answers =>
+                if (isATrust(fillingOutReturn)) {
+                  Some(AmountInPence(0))
+                } else answers.fold(_.taxableGainOrLoss, c => Some(c.taxableGainOrLoss)),
               redirectToIfNoRequiredPreviousAnswer = routes.YearToDateLiabilityController.taxableGainOrLoss()
             )
 
           case (calculatedAnswers: CalculatedYTDAnswers, _: DraftSingleDisposalReturn) =>
-            withEstimatedIncome(calculatedAnswers) { estimatedIncome =>
+            withEstimatedIncome(calculatedAnswers, isATrust(fillingOutReturn)) { estimatedIncome =>
               commonDisplayBehaviour(calculatedAnswers)(
                 form = _.fold(
                   _.hasEstimatedDetails.fold(hasEstimatedDetailsForm)(hasEstimatedDetailsForm.fill),
                   c => hasEstimatedDetailsForm.fill(c.hasEstimatedDetails)
                 )
               )(
-                page = hasEstimatedDetailsPage(_, _)
+                page = hasEstimatedDetailsPage(_, _, isATrust(fillingOutReturn))
               )(
                 requiredPreviousAnswer = { a =>
-                  if (estimatedIncome.value > 0L)
+                  if (isATrust(fillingOutReturn)) Some(AmountInPence(0))
```

Comment:
`redirectToIfNoRequiredPreviousAnswer` will have to change for trusts as well - it should be the task list route for trusts

Reply:
Done.